velox/functions/prestosql/Sequence.cpp

```diff
@@ -0,0 +1,249 @@
+        return -1;
+      } else {
+        return endVector->as<ConstantVector<int64_t>>()->valueAt(0) -
+            startVector->asFlatVector<TInput2>()->valueAt(row) + 1;
+      }
+    } else {
+      if (startVector->asFlatVector<TInput1>()->isNullAt(row) ||
+          endVector->asFlatVector<TInput2>()->isNullAt(row)) {
+        return -1;
+      } else {
+        return endVector->asFlatVector<TInput2>()->valueAt(row) -
+            startVector->asFlatVector<TInput1>()->valueAt(row) + 1;
+      }
+    }
+  }
+
+  void writeToElementsForVector(
+      long long* prt,
+      int elementsOffset,
+      VectorPtr startVector,
+      VectorPtr endVector,
+      vector_size_t row) const {
+    long start;
+    long end;
+    if (startVector->isConstantEncoding() && !endVector->isConstantEncoding()) {
+      start = startVector->as<ConstantVector<int64_t>>()->valueAt(0);
+      end = endVector->asFlatVector<TInput2>()->valueAt(row);
+    } else if (
+        !startVector->isConstantEncoding() && endVector->isConstantEncoding()) {
+      start = startVector->asFlatVector<TInput1>()->valueAt(row);
+      end = endVector->as<ConstantVector<int64_t>>()->valueAt(0);
+    } else {
+      start = startVector->asFlatVector<TInput1>()->valueAt(row);
+      end = endVector->asFlatVector<TInput2>()->valueAt(row);
+    }
+    writeToElements(prt, start, end, elementsOffset);
+  }
+
+  void writeToElements(long long* prt, long start, long end, int elementsOffset)
+      const {
+    VELOX_USER_CHECK_LE(
+        end - start,
+        kMaxResultEntries,
+        "Count of elements in a sequence must be less than or equal to " +
+            kMaxResultEntries);
+    for (auto i = start; i <= end; i++) {
+      prt[elementsOffset] = i;
+      elementsOffset++;
+    }
+  }
+};
+
+} // namespace
+
+// Create function.
+std::shared_ptr<exec::VectorFunction> create(
+    const std::string& /* name */,
+    const std::vector<exec::VectorFunctionArg>& inputArgs) {
+  auto element1Type = inputArgs.at(0).type->kind();
+  auto element2Type = inputArgs.at(1).type->kind();
+  if (element1Type == TypeKind::INTEGER && element2Type == TypeKind::INTEGER) {
+    return std::make_shared<SequenceFunction<
+        TypeTraits<TypeKind::INTEGER>::NativeType,
+        TypeTraits<TypeKind::INTEGER>::NativeType>>();
+  } else if (
+      element1Type == TypeKind::BIGINT && element2Type == TypeKind::BIGINT) {
+    return std::make_shared<SequenceFunction<
+        TypeTraits<TypeKind::BIGINT>::NativeType,
+        TypeTraits<TypeKind::BIGINT>::NativeType>>();
+  } else {
+    VELOX_FAIL("Unsupported Type")
+  }
+}
+
+std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
+  static const std::map<std::string, std::string> typePairs = {
+      {"integer", "integer"}, {"bigint", "bigint"}};
+  std::vector<std::shared_ptr<exec::FunctionSignature>> signatures;
+  signatures.reserve(typePairs.size());
+  for (const auto& [argType, arg2Type] : typePairs) {
```

Comment:
Looks like types of 2 arguments are always match. It might be simpler to write this as

```
for (const auto& type : {"integer", "bigint"}) {
   signatures.emplace_back(exec::FunctionSignatureBuilder()
                                .returnType("array(bigint)")
                                .argumentType(type)
                                .argumentType(type)
                                .build());
}
```